### ISSUE TITLE
Too many elements are missing from the document formatting options to avoid unnecessary reading for experienced users.

### ISSUE THREAD

**Issue by titet11** [nvaccess/nvda#15620]
Some of the missing items are based on the following readings:
"property page", "outer header", "data grid", "toolbar", "item list", "checkbox", "outer table", "outer edit", "panel details" , "content", "side panel", "item details", "Open chat details for", "site main menu", "multiline", "property page", "list", "grouping", " chat list", "out of edit", "out of menu".

**Comment by CyrilleB79** [comment]
Hi @titet11

Please use the [template](https://github.com/nvaccess/nvda/issues/new/choose), either bug report or new feature depending on the situation, and complete all the fields.

More specifically, we need a description of the issue you are facing and a description of what you would have expected instead before thinking to how a solution can be implemented.

**Comment by titet11** [comment]
@CyrilleB79 

This is very censillo. You simply have to add the descriptions of the elements that I mentioned above, in the section called “document format.”


There are too many descriptive elements that are unnecessary for experienced users.

One of the sections that should be able to be disabled is:

property page: Out of; datagrid, itemlist; chat list; list of background processes; cluster; box; user decline; conversation list; multiline; document; main menu; open details; submenu; It has details; message list; datagrid; item details; side panels; details panel; content; toolbar; menu button; Description of the article

**Comment by enessaribas** [comment]
I think this would be the enhancements templet rather than bug templet

**Comment by josephsl** [comment]
Hi,

As much as we would like to see how this might be done, we cannot proceed with this request without helping NVDA developers better understand why this is needed in a slightly more structured way i.e. templates (in this case, feature request), example websites where verbosity can be problematic, and so on. The "elements" listed are a combination of control roles and states, and that will require not only changing document formatting settings, but looking at control role/state announcements as well.

Something to think about (teachable moment): it would not be a good idea to insist that someone implement this, more so if the person is replying to a comment, their GitHub username appears in what's new document, and they seem to have done work that appears to be what you are looking for (including add-ons and elsewhere). It is okay to ask people to make changes if done in a way that can be understood by people reading this issue tracker - NVDA developers, NV Access, users, people from other organizations, and the general public. Please remember that this is a public venue, so think about how the way you talk about things in a GitHub issue tracker can create a public image of you that could be interpreted differently. I am telling you this early (from the perspective of a retired code contributor) to help you understand that GitHub is, after all, a repository of both code and discussions surrounding it, more so if the repository is a public one such as NVDA. There is a reason for people filling out the template - to help developers understand what's up in a more structured way (read issues and discussions I myself have created (#13544 and #14081, both implemented and closed)  and you can see how formal I can be, knowing that this is a public forum).

Hope this helps. Thanks.

**Comment by titet11** [comment]
@josephsl 
Starting with NVDA 2019.3, if I'm not mistaken, they implemented the ability to describe additional elements on each website and application.
This gives me the idea that maybe there is a way to disable all additional descriptions for a website element.

When I say additional, I do not mean those that NVDA brings by default, but rather those that they describe as such, an individual element.

This would make your job easier.

Still, when looking at the different types of elements on a website, I don't think it takes as much time to identify them as it seems.

They should try it and if it doesn't work they could make judgments about what would be more convenient.

In any case, I beg you not to abandon this idea, please.

**Comment by josephsl** [comment]
Hi,

A few more things:

1. I know you have provided examples of verbose elements, but we need an actual example such as a website that demonstrates this issue.
2. Are you sure the feature being described was introduced in NVDA 2019.3? Have you tested this with say, 2019.2.1 to see if it is indeed something introduced with 2019.3? Not all changes are documented in what's new document.
3. I'm sure you might not, but are you using any NVDA add-ons?
4. Is the behavior happening across web browsers and websites, or with a specific browser (or a family of them)?

Notice that these questions are coming from the issue template. One way we may keep the suggestion is if these questions are answered as part of the first comment (when filing issues). That way, not only it makes the issue more structured, filling out the template also improves the credibility of the issue submitter (credibility is where I'm getting at). Speaking of credibility, another way to improve submitter credibility is thinking about details before filing issues via GitHub so the submitter, code maintainers, and contributors can spend more time thinking about the issue more than asking for answers as part of our comments.

I'm known to be a bit gentle when guiding people when creating or refining GitHub issues. But I intentionally used slightly harsher tone this time to let you know what to expect in the future. I understand that GitHub might be new and filing issues for a project with specific requirements might be a new experience. Even then, one way to prepare you for further contributions (in some ways, socializing you to the culture surrounding NVDA development) is to help you see the bigger picture on why requirements exist and attention to details is crucial.

So my advice is this: before filling out the next issue, do a Google search for why developers love detailed bug reports and feature requests, and why structure is important when filing issues. If you want, send an email to NV Access (info@nvaccess.org) with your idea in hopes that you can work with NV Access folks on structuring and articulating your idea before submitting issues on GitHub.

One disclosure: I (Joseph Lee) do not work for NV Access - I was not employed by NV Access in the first place despite the way I approach the situation from the viewpoint of an insider (I rather call myself a code contributor, or, these days, I call myself a retired code contributor as I have moved onto more important things in life).

Hope this helps. Thanks.

**Comment by titet11** [comment]
@josephsl 
answering your questions:
1. Friend, what I'm telling you is very simple, and as you well know, each website has different personalized descriptions of each element. However, as noted above, NVDA was under no obligation to read these types of additional descriptions.

2. I'm not sure what version these changes were implemented in and going through the documentation will take a long time, but I know it was around 2019 or 2020 or so.
3. I use very few NVDA plugins. I only use winwizar resource monitor and virtual review.


In any case, the type of plugins you may be using is not related to adding additional elements.

**Comment by josephsl** [comment]
Hi,

To get a better idea of what's going on, we need answer to question 4 (behavior across browsers and websites). The way things are announced on websites can depend on the information exposed by different rendering engines (and accessibility API), and that can tell us how to go about possibly (not yet) resolving it. As a starting point, may I suggest testing the same website, same NVDA version, same add-ons under Microsoft Edge, Google Chrome, and Mozilla Firefox? (there is a fourth test case, but that won't be necessary unless you would like to see how Microsoft Edge may present different things if using different accessibility API's.)
Thanks.

**Comment by titet11** [comment]
josephsl 
I will give you an example.

When I'm in Windows File Explorer, the following item descriptions appear:

"item view" "expanded", "collapsed", "settings box", "tree", "status bar", "area", "divider button", "view modes"

In the case of websites, the descriptions that I mentioned in the previous comments appear.

**Comment by josephsl** [comment]
Hi,

It looks like we are walking on parallel roads... What I really asked about was behavior across different web browsers, not across programs. While a comparison across programs is fine, what I really asked for was comparisons across web browsers as things can be announced differently. The information from File Explorer window described above is presented that way because that's the information NVDA receives from File Explorer itself. You may hear words listed above from above comments from websites because that's what the web author and the browser rendering engine (and API's) think is important for screen readers to announce.

If we had the comparison information earlier (or even included when creating this issue), we wouldn't go through detours to arrive at the heart of the matter (the heart of the matter being verbosity, and based on the words listed in the first comment, looks like reducing verbosity on websites). This is why details do matter, including comparisons so people can understand what's going on and get a clear picture when asking questions. The earlier these details are given, the better, preferably the moment the issue is filed (or before then), and that's really the lesson I wanted you to learn through this exercise.

Hope this helps. Thanks.

**Comment by titet11** [comment]
josephsl 
ok, I came up with an idea to save you the work:

The idea consists of the following:

Any description that is based on an element must have an unknown sign.

This unknown sign will ensure that the words describing the item do not override any dialogue that may contain the same words.

For example. instead of being "item view", it could be: "__item view"

**Comment by titet11** [comment]
@josephsl  

That is, adding an unknown sign to the description of each item will allow each user to add words related to an item to the dictionary without affecting items that contain a description with the same words.

I hope you can understand.

**Comment by titet11** [comment]
Logically, to add a description of the element to the word dictionary, the unknown sign must be placed.

**Comment by titet11** [comment]
@josephsl 
Answering your question, I have only tried mosilla firefox and google chrome, and they both use the same descriptions on elements placed on any website.

**Comment by josephsl** [comment]
Hi,

You have answered the actual question I asked (great)! At least this should provide people with something to think about as we can somewhat narrow down the map a bit.

As for the idea proposed, perhaps. But I think there is an even simpler solution: try experimenting with regular expressions in NVDA speech dictionaries and see if that makes a difference. If it doesn't, then perhaps we can think about the proposed idea or something else.

CC @JCSTeh who works at Mozilla Foundation...

Thanks.

**Comment by titet11** [comment]
@josephsl 
All this time I have used regular expressions to override some descriptive element and yet when I write a text containing the same words, these words are overridden and therefore this solution is not reliable.

The only way to fix this problem is to place an unknown sign before any description of an item.

I also used case sensitivity, but this still overrides words that contain similar text.

In short, I have used all the options that the dictionary offers but I am completely sure that the only way to solve this problem is the solution mentioned above.

**Comment by josephsl** [comment]
Hi,

I see. What would happen if somehow the web author decides to put the "unknown sign" as part of the description, or perhaps NVDA comes across a control role/state with that unknown sign present? How can this sign be customizable for speakers of languages other than English? How are we going to take braille users into account? These are things that usually come up when someone writes a pull request, but it is helpful to think about different user experiences before committing to a solution.

Based on the discussion here, I think we might as well rename this issue, perhaps stating that we need options in document formatting settings to make some things configurable to reduce verbosity for experienced users. As I noted earlier, these verbose announcements touch both control role and state, so we need something more than just new options in document formatting settings to take the situation into account. Or, even better, I think we might as well convert this into a discussion, have some more dialogue on things, and then come up with something (I can tell you that I may not be able to work on it as I need to prioritize things other than NVDA for a while).

Thanks.

**Comment by titet11** [comment]
@josephsl 
Let's remember that there are many unknown pages and applications to create symbols, emojis or signs.

In the event that a page places our unknown sign inside an element, it would have no consequences. However, if this unknown sign is placed in any text that is not linked in some description of an item, it would be an act of revenge and would be harmful.

In these cases, what could be done is to manually add each of the descriptions found for any program and website.

Remember that previous versions of NVDA did not require describing custom elements on a website, so you must read the documentation to create a function that overrides any custom description of an element on any website.

**Comment by titet11** [comment]
In short, while adding an unknown sign sounds nice, it could be used as an act of revenge, so it's best to find a way to override custom descriptions for any items.

To do this, as I said before, it is necessary to read the documentation to know from which version the possibility of reading personalized descriptions of each element was not allowed.

**Comment by josephsl** [comment]
Hi,

If I understand you right, I think you are referring to a change introduced in NVDA 2020.4 where NVDA announces aria role description text coded by web authors. This points to an ARIA standard, which NVDA tries to faithfully implement, and so do web browsers and their rendering engines. This really points to web authors' understandings or misunderstandings as a possible issue, which changes the picture and the overall thread. I'll leave it up to you to figure out why and a possible solution that might not involve adding many settings at once (and to test my theory, try NVDA 2020.3 and 2020.4 and see if that's the behavior you are expecting from the former).

Thanks.

**Comment by seanbudd** [comment]
Welcome @titet11 

We are unable to process this issue as it stands.

Please fill out the issue template fully.

Kind Regards,
NV Access Software Developers

**Comment by titet11** [comment]
@seanbudd 
Look, I have already explained the problem in detail with all the doubts you may have, so don't say that you can't process it.

**Comment by josephsl** [comment]
Hi,

Simply put, issue closed as it is not structured, as I and others have advised in earlier comments.

I think it was pointed out in another post that English may not be your first language. This is okay, but it does not take away the need to think about and follow (if possible) the advice given by others, more so on a public repository like NVDA screen reader project.

I would like to take this time to give you a more serious (albeit kindly) public warning/advice: partnership and situational awareness matter. GitHub is not a chat room or an informal venue where you can just talk about what you want. There is a reason for me and others advising you that you need to answer questions we asked, preferably as part of filing an issue via issue/feature request template. Remember my comment from a few hours ago where I said that you answered the question I asked and added an exclamation point. That's the feeling we (developers) get when we do see someone filling out the issue template and answering included questions. To be honest with you, I was "masking" (hiding my true emotion to respond appropriately to a situation) my true feelings: frustration and anger (I'm known in the community (NVDA project and elsewhere) to wait patiently and then explode all at once, and I was mentally experiencing this while we walked on parallel roads (which eventually intersected)). A somewhat related term is emotional labor where a developer would use emotions to show (or hide) our professional selves when talking in public, including when asking questions that can help folks understand what's going on. Situational awareness also falls upon the issue submitter, including using tools provided by the project to file issues and trying to communicate in a way that helps developers get a better picture from the start.

What I am about to say may sound harsh or hard to understand at the moment, but it is important to talk about it before it is too late: your credibility is at stake. Despite advice to use the issue template, you have now raised two issues (subsequently closed) without use of a template. Despite calls to answer the question asked by community members, you have dodged around them. Despite a response from NV Access staff telling you the reason for issue closure, you insist that you have provided everything asked for, implying that comments should be sufficient to justify keeping the issue alive. You can say that my misunderstandings caused this issue, and I have apologized about that.

However, as a member of NVDA project (a retired code contributor), I am sorry to tell you this in a serious tone, and to do so with sadness and disappointment on a public venue: you have lost me and your credibility for the time being. How can (and do) you get me and your credibility back? Take the advice given by people here, starting with filling out the template. I would also like to add one more thing to this: think for some time before opening a new GitHub issue - think about the questions I and others asked, think about possibilities across aps and situations, think about possible solutions (if any), and think about how you can respond to others. I'm giving you this personal advice not only as an NVDA community member, but thinking before posting is one way to reduce developer frustrations.

So please remember and do as I advise: starting from your third GitHub issue, I, NV Access staff, and NVDA project community members would like to see that you did fill out the template. You might try and just leave the template answers blank and add a second comment with your issue/bug report/suggestion text - don't do that. Do not force us to travel on parallel roads again.

Hope this helps. Thanks.

**Comment by titet11** [comment]
@josephsl 
I understand that I did not do things correctly, and it is not because I did not want to, but because I did not know how to find the questions that they talk to me so much about:
However, I just found them.

Regardless of this, both problems that I mentioned above are almost solved except this problem since I need to investigate in which version of NVDA custom descriptions were added for each element on a website, although these types of investigations should not be my obligation since they are the developers and you are the ones who know how you are making the changes.

Anyway, I don't know what they gain by being so dramatic as to say that they can't fix this problem because of a couple of questions that were basically already answered very precisely in the comments.
Even more ridiculous is that it's not even a complex topic to understand, so I can't understand the need to try to pretend that they can't understand this problem just because I didn't directly answer the requested template, which basically, already has the questions answered in the comments.

You as developers should be able to understand this very simple problem I mentioned.

I understand that if it were a more complicated topic it would be difficult to understand, but the question is, what is so complicated about understanding that we simply need to apply a solution to avoid unnecessary descriptions in each element of a website? or application. Do you have any major complications in understanding this simple and necessary problem?

**Comment by josephsl** [comment]
Hi,

It is not the question of complexity of a problem. The overall message from NV Access and contributors (including myself even if came across as dramatic at times) is helping you see the value of structuring things, and that's where folks were going when talking about issue template.

Also, looking for changes across versions is something the template will ask you, so it means it is also up to the issue submitter to do some work. You see, when it comes to software development, it is not just developers who must do all the work - a product is shaped by an understanding between developers and users, with users also playing a part via investigations as well. We can also say that user testing is a form of helping developers understand what's going on, more so when developers come up with something based on user feedback, and users trying out things so both parties (and countless others) can verify that the fix is indeed a fix. This is why I asked about testing with older NVDA releases and reporting back so that: developers can do some more thinking, and users to test a possible fix (or two) in the future.

Simply adding comments to an issue does not keep an issue alive or motivate developers to work on things. As you saw, it is also important to offer things in a way that can be understood by different people, more so if asked to do it in a more structured way. And doing things in a more structured way, especially when asked by community members and the organization in charge of overall NVDA development, is one way to improve credibility.

As a personal note: notice the tone I'm using to explain all this. You did see my warning a few minutes ago, and that's perhaps the maximum display of emotion you can see from me throughout NVDA project (look at other GitHub issues where I have written things and you can see why). I'm showing gentleness and go to lengths to explain community positions in a way that's understandable to new people. You may also notice that I take a "teaching" attitude when talking about all this because there are some life lessons for us to learn based on what we are talking about throughout this GitHub issue thread.

So let me offer a choice: we can create a brand new issue complete with the issue template filled in, discuss the closed issue here with more comments, or take a short break from filing issues on GitHub. These options will have short-term and long-term outcomes (personally and for the community), so choose wisely.

Thanks.

**Comment by titet11** [comment]
@josephsl 
I'm too lazy to make a new article just so it doesn't diminish your interest in something that is important to any user, but maybe one of these days I will.
